common/types/bytes.go

```diff
@@ -58,7 +58,14 @@ func (b Bytes) Compare(other ref.Val) ref.Val {
 // ConvertToNative implements the ref.Val interface method.
 func (b Bytes) ConvertToNative(typeDesc reflect.Type) (any, error) {
 	switch typeDesc.Kind() {
-	case reflect.Array, reflect.Slice:
+	case reflect.Array:
```

Comment:
Why is the custom conversion needed here? 

I've tested this on the go playground and it should be possible to convert from a byte slice to an byte array, as long as the byte slice has more items than the requested array.

https://go.dev/play/p/fxZM4NPhWD8?v=goprev

Reply:
There seems to be some issue with converting between byte and uint8 in the Bazel builds. I'm not sure why, but this change works around the problem

Reply:
There should probably be a check to ensure that the `len` of the requested array is equals to the `len` of the returned array. Otherwise this would fail on type conversion.

e.g. 
```go
b := types.Bytes([]byte{1})
val, _ := b.ConvertToNative(reflect.TypeOf([10]byte{})
val.([10]byte) // <- this will panic
```

Reply:
Probably 
`refArrPtr := reflect.New(reflect.ArrayOf(len(b), typeDesc.Elem()))`
can be changed to 
`refArrPtr := reflect.New(typeDesc.Elem)`

but the sanity check for `len` should still be put in place imho

Reply:
Thanks for the catch!